docs/unified-test-documentation/dasharo-compatibility/364-p-e-cores.md

```diff
@@ -0,0 +1,588 @@
+# Dasharo Compatibility: CPU cores enabling and disabling
+
+## Test cases common documentation
+
+**Important notice**
+
+If both are technically supported, as is the case for NovaCustom NV4x, the
+test should be carried out separately for both the default Intel network card
+and the Atheros one.
+
+**Test setup**
+
+1. Proceed with the
+    [Generic test setup: firmware](../generic-test-setup.md#firmware).
+1. Proceed with the
+    [Generic test setup: OS installer](../generic-test-setup.md#os-installer).
+1. Proceed with the
+    [Generic test setup: OS installation](../generic-test-setup.md#os-installation).
+1. Proceed with the
+    [Generic test setup: OS boot from disk](../generic-test-setup.md#os-boot-from-disk).
+
+## CCC001.001 CPU Hyper-Threading disabled (Ubuntu)
+
+**Test description**
+
+This test aims to verify that the Hyper-Threading can be disabled and the change
+can be detected from the operating system.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+
+**Test steps**
+
+1. Power on the DUT.
+1. While the DUT is booting, hold the `BIOS_SETUP_KEY` to enter the UEFI Setup
+    Menu.
+1. Enter the `Dasharo System Features` menu using the arrow keys and Enter.
+1. Enter the `CPU Configuration` submenu.
+1. Verify that the `Hyper-Threading` option is not chosen - if
+    so, press `Space` and then `F10` to save the changes.
+1. If necessary - press `Y` to confirm saving the changes.
+1. Go back to the main menu using the `ESC` key.
+1. Select the `Reset` option to apply the settings and reboot.
+1. Boot into the system.
+1. Log into the system by using the proper login and password.
+1. Detect or install a required os package: 'util-linux'
+1. Execute the following command in the terminal to check platform core count
+    conditions:
+
+    ```bash
+    lscpu | grep "Thread(s) per core: "
+    ```
+
+1. Compare the number with expected value.
+
+**Expected result**
+
+For disabled Hyper-Threading expected value is 1.
```

Comment:
Yes, for non-hybrid it is of course enough, but that's only for P-cores in a hybrid system. E-cores always have 1 thread per core. The verification method is not sufficient as is ambiguous depending on core type for hybrid system.

Reply:
@miczyg1 Would it be ok to disable this test by default in machines that have the same number of cores as threads? Such DUTs shouldnt have HT at all?

Reply:
And how would you detect that number of threads and cores is the same? Note that with HT disabled, the number of cores and threads will be the same. How can you be sure that you run with HT disabled vs HT not supported?

Reply:
Simply have the HT flag whether the given test should be run on the given platform. As we do for all of the other features.

Reply:
Actually configs already do have this information stored in ```${DEF_THREADS_PER_CORE}=            1``` key.
I could use it like in: b976dd14b674385561910dc600f45a5a09214ad9
Or I could add HT flag to each config if it would be better.